easybuild/tools/modules.py

```diff
@@ -750,6 +750,14 @@ def modpath_extensions_for(self, mod_names):
             modtxt = self.read_module_file(mod_name)
             exts = [ext for tup in modpath_ext_regex.findall(modtxt) for ext in tup if ext]
             self.log.debug("Found $MODULEPATH extensions for %s: %s", mod_name, exts)
+            if exts:
+                modtxt = self.show(mod_name)
```

Comment:
include comment to explain why you're going through `show`, since you already have the list of `$MODULEPATH` extensions?